gtk4/src/subclass/entry_buffer.rs

```diff
@@ -0,0 +1,233 @@
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkEntryBufferClass;
+            let f = (*parent_class)
+                .insert_text
+                .expect("No parent class impl for \"insert_text\"");
+
+            f(
+                entry_buffer
+                    .unsafe_cast_ref::<EntryBuffer>()
+                    .to_glib_none()
+                    .0,
+                position,
+                chars.to_glib_none().0,
+                chars.len() as u32,
+            )
+        }
+    }
+
+    fn parent_inserted_text(&self, entry_buffer: &Self::Type, position: u32, chars: &str) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkEntryBufferClass;
+            if let Some(f) = (*parent_class).inserted_text {
+                f(
+                    entry_buffer
+                        .unsafe_cast_ref::<EntryBuffer>()
+                        .to_glib_none()
+                        .0,
+                    position,
+                    chars.to_glib_none().0,
+                    chars.len() as u32,
+                )
+            }
+        }
+    }
+}
+
+unsafe impl<T: EntryBufferImpl> IsSubclassable<T> for EntryBuffer {
+    fn override_vfuncs(class: &mut glib::Class<Self>) {
+        <Object as IsSubclassable<T>>::override_vfuncs(class);
+
+        let klass = class.as_mut();
+        klass.delete_text = Some(entry_buffer_delete_text::<T>);
+        klass.deleted_text = Some(entry_buffer_deleted_text::<T>);
+        klass.get_length = Some(entry_buffer_get_length::<T>);
+        klass.get_text = Some(entry_buffer_get_text::<T>);
+        klass.insert_text = Some(entry_buffer_insert_text::<T>);
+        klass.inserted_text = Some(entry_buffer_inserted_text::<T>);
+    }
+}
+
+unsafe extern "C" fn entry_buffer_delete_text<T: EntryBufferImpl>(
+    ptr: *mut ffi::GtkEntryBuffer,
+    position: u32,
+    n_chars: u32,
+) -> u32 {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<EntryBuffer> = from_glib_borrow(ptr);
+
+    imp.delete_text(wrap.unsafe_cast_ref(), position, n_chars)
+}
+
+unsafe extern "C" fn entry_buffer_deleted_text<T: EntryBufferImpl>(
+    ptr: *mut ffi::GtkEntryBuffer,
+    position: u32,
+    n_chars: u32,
+) {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<EntryBuffer> = from_glib_borrow(ptr);
+
+    imp.deleted_text(wrap.unsafe_cast_ref(), position, n_chars)
+}
+
+unsafe extern "C" fn entry_buffer_get_text<T: EntryBufferImpl>(
+    ptr: *mut ffi::GtkEntryBuffer,
+    n_bytes: *mut usize,
+) -> *const libc::c_char {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<EntryBuffer> = from_glib_borrow(ptr);
+
+    imp.get_text(wrap.unsafe_cast_ref(), &mut (n_bytes as usize))
```

Comment:
Not sure if `n_bytes as usize` is right way to do cast my pointer to a usize?

Reply:
No, see above. You'd get back a `GString` from `get_text()` and then return the pointer to it and store the length in `n_bytes`.

It's an out parameter for the length of the string. What you do right now is unsound and you also pass an uninitialized value around :)